lib/keila_web/views/public_form_view.ex

```diff
@@ -345,13 +345,91 @@ defmodule KeilaWeb.PublicFormView do
+  end
+
+  def render_unsubscribe_form(form, unsubscribed, assigns) do
     form_styles = build_form_styles(form)
 
     content_tag(:div, class: @form_classes, style: form_styles) do
-      gettext("You have been unsubscribed from this list.")
+      if unsubscribed do
+        content_tag(:div, class: "text-center") do
+          [
+            content_tag(:h1, gettext("Unsubscribed"), class: "text-2xl font-bold mb-4"),
+            content_tag(:p, gettext("You have been unsubscribed from this list."), class: "text-lg")
+          ]
+        end
+      else
+        content_tag(:div, class: "text-center") do
+          [
+            content_tag(:h1, gettext("Unsubscribe"), class: "text-2xl font-bold mb-4"),
+            content_tag(:p, gettext("Are you sure you want to unsubscribe from this list?"), class: "text-lg mb-6"),
+            render_unsubscribe_button_form(assigns)
+          ]
+        end
+      end
     end
   end
 
+  defp render_unsubscribe_button_form(assigns) do
+    action_url = 
+      if Map.get(assigns, :deprecated_route) do
+        Routes.public_form_url(KeilaWeb.Endpoint, :unsubscribe, assigns.project_id, assigns.contact_id)
+      else
+        Routes.public_form_url(KeilaWeb.Endpoint, :unsubscribe, assigns.project_id, assigns.recipient_id, assigns.hmac)
+      end
+
+    form_tag(action_url, [method: :post, id: "unsubscribe-form", "data-requires-js": "true"]) do
+      [
+        content_tag(:div, "", [id: "js-check", style: "display: none;"]),
+        content_tag(:noscript) do
+          content_tag(:p, gettext("JavaScript is required to unsubscribe. Please enable JavaScript and try again."), 
+            class: "text-red-600 mb-4")
+        end,
+        hidden_input(:form, "confirm_unsubscribe", value: "true"),
+        content_tag(:div, class: "space-y-4") do
+          [
+            content_tag(:button, gettext("Unsubscribe"),
+              [
+                type: "button",
+                id: "unsubscribe-btn",
+                class: "bg-red-600 hover:bg-red-700 text-white font-bold py-3 px-6 rounded cursor-pointer disabled:opacity-50 disabled:cursor-not-allowed",
+                onclick: "handleUnsubscribe()"
+              ]
+            ),
+            content_tag(:script, raw(unsubscribe_javascript()), type: "text/javascript")
+          ]
+        end
+      ]
+    end
+  end
+
+  defp unsubscribe_javascript do
+    """
+    function handleUnsubscribe() {
+      // Check if JavaScript is enabled
+      document.getElementById('js-check').style.display = 'block';
+      
+      // Show confirmation dialog
+      if (confirm('#{String.replace(gettext("Are you sure you want to unsubscribe?"), "'", "\\'")}')) {
+        var btn = document.getElementById('unsubscribe-btn');
+        btn.disabled = true;
+        btn.textContent = '#{String.replace(gettext("Unsubscribing..."), "'", "\\'")}';
+        
+        // Add a small delay to prevent immediate automated requests
```

Comment:
What exactly is the idea here? That automated visits might click on the button but then not wait 500ms?